lib/services/livesync/livesync-service.ts

```diff
@@ -527,7 +527,7 @@ export class LiveSyncService extends EventEmitter implements IDebugLiveSyncServi
 
 		const currentWatcherInfo = this.liveSyncProcessesInfo[liveSyncData.projectDir].watcherInfo;
 
-		if (!currentWatcherInfo || currentWatcherInfo.pattern !== pattern) {
+		if (!currentWatcherInfo || currentWatcherInfo.pattern.toString() !== pattern.toString()) {
```

Comment:
It might be good to sort the array (if its an array) so that the order of patterns doesn't matter.

Reply:
Actually, `pattern` is always an array as it can be seen [here](https://github.com/NativeScript/nativescript-cli/blob/ab4e6e43d7be6e3d7b5aff5e1d83cf8a9be333db/lib/services/livesync/livesync-service.ts#L516). `currentWatcherInfo.pattern` on the other hand is always an array too as [it is actually the same `pattern` ](https://github.com/NativeScript/nativescript-cli/blob/ab4e6e43d7be6e3d7b5aff5e1d83cf8a9be333db/lib/services/livesync/livesync-service.ts#L653).

So there is no need to sort the array, as both arrays always have the same order.

@KristianDD what do you think?

Reply:
But as seen [here](https://github.com/NativeScript/nativescript-cli/blob/013ed913482a441f30c24abfcae98a34dc6a0dd9/lib/definitions/livesync.d.ts#L63) it can be a string. And as I will add configurations for livesync the patterns might change.